xbmc/video/VideoDatabase.cpp

```diff
@@ -3934,6 +4039,46 @@ void CVideoDatabase::GetCast(int media_id, const std::string &media_type, std::v
   }
 }
 
+void CVideoDatabase::GetCast(const std::string& media_ids, const std::string &media_type, std::map<int, CVideoInfoTag>& details)
```

Comment:
Without having seen the other solution I think that would a lot better, yes. I see this string as a low level query detail for each of these methods, and keeping it contained here would remove the added complexity of constructing/keeping track of it in the `Get*ByWhere` methods.
I wouldn't worry about any overhead of creating it again. Should be practically free compared to everything else going on here:)